src/aggregator/aggregator/aggregator.go

```diff
@@ -313,19 +316,41 @@ func (agg *aggregator) processPlacementWithLock(
 	if !agg.shouldProcessPlacementWithLock(newStagedPlacement, newPlacement) {
 		return nil
 	}
-	var newShardSet shard.Shards
+	var (
+		notFoundInPlacement = false
+		newShardSet         shard.Shards
+	)
 	instance, err := agg.placementManager.InstanceFrom(newPlacement)
 	if err == nil {
 		newShardSet = instance.Shards()
 	} else if err == ErrInstanceNotFoundInPlacement {
+		notFoundInPlacement = true
 		newShardSet = shard.NewShards(nil)
 	} else {
 		return err
 	}
+
 	agg.updateShardsWithLock(newStagedPlacement, newPlacement, newShardSet)
 	if err := agg.updateShardSetIDWithLock(instance); err != nil {
 		return err
 	}
+
+	if notFoundInPlacement {
+		// NB(r): Without this log message it's hard for operators to debug
+		// logs about receiving metrics that the aggregator does not own.
+		placementInstances := newPlacement.Instances()
+		placementInstanceIDs := make([]string, 0, len(placementInstances))
+		for _, instance := range placementInstances {
+			placementInstanceIDs = append(placementInstanceIDs, instance.ID())
+		}
+
+		msg := "aggregator instance ID must appear in placement: " +
+			"not found with current instance ID"
+		agg.logger.Error(msg,
+			zap.String("currInstanceID", agg.placementManager.InstanceID()),
+			zap.Strings("placementInstanceIDs", placementInstanceIDs))
+	}
+
```

Comment:
metagnit: is it possible to replace line 327 with this and remove the bool?

Reply:
Possible, I just wanted to only emit it though if we didn't hit some other error condition.

Reply:
Had another look, I'll move it and remove the bool, I'm adding some context to the message so it makes sense to emit the log from that location `"no shards assigned since not found with current instance ID"`